server.js

```diff
@@ -13,8 +14,126 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+//reading all bookings
+app.get("/bookings", (req, res) => {
+  res.status(200).send(bookings);
+});
+
+app.post("/bookings", (req, res) => {
+  const requestBody = req.body;
+
+  const newBooking = {
+    id: requestBody.id,
+    title: requestBody.title,
+    firstName: requestBody.firstName,
+    surname: requestBody.surname,
+    email: requestBody.email,
+    roomId: requestBody.roomId,
+    checkInDate: requestBody.checkInDate,
+    checkOutDate: requestBody.checkOutDate,
+  };
+
+  // validation
+
+  const validBooking =
```

Comment:
This works but I don't really like the way this validation is done, it's a bit of a pain to read and manage. I would have been inclined to used a loop to check that none of the properties are empty and do the special cases like email validation afterwards, but this is a matter of personal taste

Reply:
got that👍